File: 2020.8/旷视-等差数列.cpp
#include <iostream>
#include <cstdio>
#include <vector>
#include <algorithm>
using namespace std;

int main() {
    while(true) {
        int n, tmp;
        cin >> n;
        vector<int> array;
        for (int i = 0; i < n; i++) {
            cin >> tmp;
            array.push_back(tmp);
        }
        sort(array.begin(), array.end());
        int magicNums = 0;
        while (!array.empty() && array[magicNums] == 0)
            magicNums++;
        int i = magicNums + 1;
        bool vaild = true;
        while(i < array.size()) {
            if(array[i] == array[i-1]) {
                vaild = false;
                break;
            }
            if(array[i] - array[i - 1] > 1) {
                int need = array[i] - array[i - 1] - 1;
                if (magicNums >= need)
                    magicNums -= need;
                else
                {
                    vaild = false;
                    break;
                }
            }
            i++;
        }
        if (vaild)
            cout << "Valid"<<endl;
        else
            cout<<"Invalid"<<endl;
    }
    return 0;
}

File: 2020.8/旷视-等差数列.py
import sys


def _input():
    return sys.stdin.readline()


while True:
    N = int(_input().strip())
    array = list(map(int, _input().strip().split()))[:N]
    array.sort()
    magicNums = 0
    while array and array[magicNums] == 0:
        magicNums += 1
    i = magicNums + 1
    while i < len(array):
        if array[i] == array[i - 1]:
            break
        if array[i] - array[i - 1] > 1:
            need = array[i] - array[i - 1] - 1
            if magicNums >= need:
                magicNums -= need
            else:
                break
        i += 1
    else:
        print('Valid')
        continue
    print('Invalid')
